modules/boss.lua

```diff
@@ -39,22 +79,33 @@ function mod:PLAYER_REGEN_ENABLED()
 end
 
 function mod:DebugTest()
-  BossAttempt("BossKilled", "Sapphiron")
-  BossAttempt("kill", "Bob")
-  BossAttempt("wipe", "Spongebob")
+  NewBossAttempt("BossKilled", 715)
 end
 
-mod.dbDefaults = {
-  profile = {
-    enabled = false,
-    wipedetection = false,
-  },
-}
-
 function mod:OnInitialize()
   self.db = EPGP.db:RegisterNamespace("boss", mod.dbDefaults)
 end
 
+function mod:GetEncounterEP(encounter_id)
+  if mod.db.profile.bossreward[encounter_id] ~= nil then
```

Comment:
As `mod.db.profile.bossreward` is initialized early on as `{}` or a filled table with configured values from previous sessions, this will work fine without any extra validation. This will return the configured value or 0.

It doesn't matter what `encounter_id` is, it will always return 0 or a configured value if `od.db.profile.bossreward[encounter_id]` is valid.

Since `EncounterAttempt` was updated with better input validation, `encounter_id` here will always be a valid value.